File: testdata/jennies/builders/nullable_map_assignment/PythonBuilder/builders/nullable_map_assignment.py
import typing
from ..cog import builder as cogbuilder
from ..models import nullable_map_assignment


class SomeStruct(cogbuilder.Builder[nullable_map_assignment.SomeStruct]):
    _internal: nullable_map_assignment.SomeStruct

    def __init__(self):
        self._internal = nullable_map_assignment.SomeStruct()

    def build(self) -> nullable_map_assignment.SomeStruct:
        """
        Builds the object.
        """
        return self._internal    
    
    def config(self, config: dict[str, str]) -> typing.Self:    
        self._internal.config = config
    
        return self
    

File: testdata/jennies/builders/nullable_map_assignment/RustBuilder/src/builders/nullable_map_assignment.rs
use crate::cog;
use crate::types::nullable_map_assignment;
use std::collections::HashMap;

#[derive(Debug, Clone)]
pub struct SomeStructBuilder {
    internal: nullable_map_assignment::SomeStruct,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl SomeStructBuilder {
    pub fn new() -> Self {
        Self {
            internal: nullable_map_assignment::SomeStruct::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for SomeStructBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl SomeStructBuilder {
    pub fn config(mut self, config: HashMap<String, String>) -> Self {
        self.internal.config = config;

        self
    }
}

impl cog::Builder<nullable_map_assignment::SomeStruct> for SomeStructBuilder {
    fn build(&self) -> Result<nullable_map_assignment::SomeStruct, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}
